api/query_test.go

```diff
@@ -0,0 +1,330 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"mime"
+	"mime/multipart"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/gin-gonic/gin"
+)
+
+const well_known = "../testdata/wellknown/well_known_default.vds"
+
+type SliceTest struct {
+	name           string
+	method         string
+	slice          TestSliceQuery
+	sliceQuery     string
+	expectedStatus int
+	expectedError  string
+}
+
+// define own help types to assure separation between production and test code
+type TestErrorInfo struct {
+	Error string `json:"error" binding:"required"`
+}
+
+type TestSliceQuery struct {
+	Vds       string `json:"vds"`
+	Direction string `json:"direction"`
+	Lineno    int    `json:"lineno"`
+	Sas       string `json:"sas"`
+}
+
+func TestSliceHTTPResponse(t *testing.T) {
+	testcases := []SliceTest{
+		{
+			name:   "Valid GET Query",
+			method: http.MethodGet,
+			slice: TestSliceQuery{
+				Vds:       well_known,
+				Direction: "i",
+				Lineno:    0, //side-effect assurance that 0 is accepted
+				Sas:       "n/a",
+			},
```

Comment:
Would it make sense to have a http.Request field here in stead?
```
request: makeRequest('get', '/slice', well_known, SliceQuery{...} )
```

i.e. something similar to the prepare function. You can even capture both slice and fence queries  (with `interface{}`)  to avoid have 2 separate functions like you do now.